packages/gitbook/src/components/AdminToolbar/AdminToolbarClient.tsx

```
@@ -17,48 +19,130 @@ import {
     ToolbarSubtitle,
     ToolbarTitle,
 } from './Toolbar';
-import type { AdminToolbarClientProps } from './types';
+import {
+    type ToolbarControlsContextValue,
+    ToolbarControlsProvider,
+} from './ToolbarControlsContext';
+import type { AdminToolbarClientProps, AdminToolbarContext } from './types';
 
 export function AdminToolbarClient(props: AdminToolbarClientProps) {
-    const { context } = props;
+    const { context, onPersistentClose, onSessionClose, onToggleMinify } = props;
+    const [minified, setMinified] = React.useState(true);
     const visitorSession = useVisitorSession();
+    const [sessionClosed, setSessionClosed] = React.useState(false);
+    const [shouldHide, setShouldHide] = React.useState(false);
+
+    React.useEffect(() => {
+        const STORAGE_KEY = 'gitbook_toolbar_closed';
+
+        try {
+            const hidden = !!localStorage.getItem(STORAGE_KEY);
```

Shouldn't we use `localStorage` from `@gitbook/util-local-storage`?

---

Or how does it work on GBO? We let it crash?